etl/steps/data/garden/wb/2025-04-08/world_bank_pip_legacy.py

```diff
@@ -0,0 +1,1465 @@
+    )
+
+    # Remove columns not needed: stacked, above, etc
+    drop_list = ["above", "between", "poverty_severity", "watts"]
+    for var in drop_list:
+        tb = tb[tb.columns.drop(list(tb.filter(like=var)))]
+
+    vars = [
+        i
+        for i in tb.columns
+        if i
+        not in [
+            "country",
+            "year",
+            "ppp_version",
+            "reporting_level",
+            "welfare_type",
+            "reporting_pop",
+            "is_interpolated",
+            "distribution_type",
+            "estimation_type",
+            "survey_comparability",
+            "comparable_spell",
+            f"headcount_{INTERNATIONAL_POVERTY_LINE_CURRENT}_regions",
+            "surveys_past_decade",
+        ]
+    ]
+
+    # Define spell table list
+    spell_tables = []
+
+    # Loop over the variables in the main dataset
+    for select_var in vars:
+        tb_var = tb[["country", "year", select_var, "survey_comparability"]].copy()
+
+        # convert to wide
+        tb_var = pr.pivot(
+            tb_var,
+            index=["country", "year"],
+            columns=["survey_comparability"],
+            values=select_var,
+        )
+
+        tb_var.metadata.short_name = f"{tb_var.metadata.short_name}_{select_var}"
+
+        spell_tables.append(tb_var)
+
+    return spell_tables
+
+
+def create_survey_spells_inc_cons(tb_inc: Table, tb_cons: Table) -> list:
+    """
+    Create table for each indicator and survey spells, to be able to graph them in explorers.
+    This version recombines income and consumption tables to not lose dropped rows.
+    """
+
+    tb_inc = tb_inc.reset_index().copy()
+    tb_cons = tb_cons.reset_index().copy()
+
+    # Concatenate the two tables
+    tb_inc_or_cons_spells = pr.concat(
+        [tb_inc, tb_cons], ignore_index=True, short_name=f"income_consumption_{PPP_YEAR_CURRENT}"
+    )
+
+    # Set index and sort
+    tb_inc_or_cons_spells = tb_inc_or_cons_spells.format(keys=["country", "year", "reporting_level", "welfare_type"])
+
+    # Create spells
+    spell_tables = create_survey_spells(tb_inc_or_cons_spells)
+
+    return spell_tables
+
+
+def combine_tables_ppp_comparison(tb_ppp_old: Table, tb_ppp_current: Table, short_name: str) -> Table:
+    """
+    Combine income and consumption tables from PPP_YEAR_OLD and PPP_YEAR_CURRENT PPPs.
+    We will use this table for the Poverty Data Explorer: World Bank data - PPP_YEAR_OLD vs. PPP_YEAR_CURRENT prices.
+    """
+
+    # Identify columns to use (ID + indicators)
+    id_cols = ["country", "year"]
+
+    tb_ppp_old = define_columns_for_ppp_comparison(tb=tb_ppp_old, id_cols=id_cols, ppp_version=PPP_YEAR_OLD)
+    tb_ppp_current = define_columns_for_ppp_comparison(tb=tb_ppp_current, id_cols=id_cols, ppp_version=PPP_YEAR_CURRENT)
+
+    # Rename all the non-id columns with the suffix _ppp(year)
+    # (the suffix option in merge only adds suffix when columns coincide)
+    tb_ppp_old = tb_ppp_old.rename(
```

Comment:
When using `drop` and `rename` programmatically, it's convenient to use `errors="raise"` for safety. Otherwise, explicitly use `errors="ignore"` if that's what you need in that case.